### PR TITLE
Merge inheritance margin feature

### PR DESCRIPTION
**Pull request by Cosifne**

### CODE REVIEW

**Comment by Cosifne** [comment]
: ) looks like there are conflicts. I'll first create a resolving PR then update this

**Comment by Cosifne** [comment]
This includes the inheritance margin feature and needs M2 approval.
@vatsalyaagrawal

**Comment by Cosifne** [comment]
@sharwell @CyrusNajmabadi @ryzngard 
I need one approval to check this in, could anyone help me?

**Comment by sharwell** [comment]
I retargeted to vs-deps due to dependency changes.

File: src/EditorFeatures/CSharp/InheritanceMargin/CSharpInheritanceMarginService.cs
Comment (sharwell) [inline src/EditorFeatures/CSharp/InheritanceMargin/CSharpInheritanceMarginService.cs:44]: ❔ Missing `IndexerDeclaration`?

File: src/EditorFeatures/Core.Wpf/InheritanceMargin/MarginGlyph/InheritanceMarginViewModel.cs
Comment (sharwell) [inline src/EditorFeatures/Core.Wpf/InheritanceMargin/MarginGlyph/InheritanceMarginViewModel.cs:63]: ⚠️ This makes assumptions about the grammar structure in the language. Better to have a dedicated resource with placeholders.

File: src/EditorFeatures/Core.Wpf/EditorFeaturesWpfResources.resx
Comment (sharwell) [inline src/EditorFeatures/Core.Wpf/EditorFeaturesWpfResources.resx:197]: 💡 Would be good to have a comment that this resource is only used for UI Automation / accessibility

File: src/EditorFeatures/Core.Wpf/InheritanceMargin/InheritanceGlyphFactoryProvider.cs
Comment (sharwell) [inline src/EditorFeatures/Core.Wpf/InheritanceMargin/InheritanceGlyphFactoryProvider.cs:44]: 💡 I'd prefer this be lazily evaluated in `GetGlyphFactory`

File: src/EditorFeatures/Core.Wpf/InheritanceMargin/MarginGlyph/InheritanceMargin.xaml.cs
Comment (sharwell) [inline src/EditorFeatures/Core.Wpf/InheritanceMargin/MarginGlyph/InheritanceMargin.xaml.cs:86]:
```suggestion
            SetResourceReference(BackgroundProperty, VsBrushes.CommandBarMenuBackgroundGradientKey);
            SetResourceReference(BorderBrushProperty, VsBrushes.CommandBarMenuBorderKey);
```

File: src/EditorFeatures/Core.Wpf/InheritanceMargin/MarginGlyph/InheritanceMarginViewModel.cs
Comment (sharwell) [inline src/EditorFeatures/Core.Wpf/InheritanceMargin/MarginGlyph/InheritanceMarginViewModel.cs:90]: ⚠️ This makes assumptions about the language. Some translations prefer `"` to `'`.

File: src/EditorFeatures/Core.Wpf/Microsoft.CodeAnalysis.EditorFeatures.Wpf.csproj
Comment (sharwell) [inline src/EditorFeatures/Core.Wpf/Microsoft.CodeAnalysis.EditorFeatures.Wpf.csproj:40]: ❔ Shouldn't code depending on this should move up to the VS layer

File: src/EditorFeatures/Core/FindUsages/IDefinitionsAndReferencesFactory.cs
Comment (sharwell) [inline src/EditorFeatures/Core/FindUsages/IDefinitionsAndReferencesFactory.cs:80]: ❔ What does `FindReferencesSearchOptions` get used for here? Is there any reason why we use `FindReferencesSearchOptions.Default` without setting `UnidirectionalHierarchyCascade` (the latter performs better)

File: src/EditorFeatures/Core/GoToDefinition/GoToDefinitionHelpers.cs
Comment (sharwell) [inline src/EditorFeatures/Core/GoToDefinition/GoToDefinitionHelpers.cs:133]: 💡 After this merges, can remove the previous `TryGoToDefinition` method as a separate follow-up. You have the option of either filing a follow-up issue or submitting a PR to just remove it in favor of the new method.

**Comment by vatsalyaagrawal** [comment]
@allisonchou headsup for 16.10.P2

**Comment by Cosifne** [comment]
This PR should be ready when https://github.com/dotnet/roslyn/pull/52379 is checked in

**Review by sharwell — APPROVED** [review]

**Pull request merged**
